### ISSUE TITLE
extract_datetime_en() fails silently in several cases, returns wrong datetime

### ISSUE THREAD

**Issue by ChanceNCounter** [MycroftAI/mycroft-core#2369]
https://github.com/MycroftAI/mycroft-core/blob/a3e35d736b0426f33b7fc1d1883be38e4b650b0b/mycroft/util/lang/parse_en.py#L667

Several cases not yet implemented and consequently parsing incorrectly:

- [ ]  **"Yesterday**: this one's probably somewhat urgent - may constitute a bug rather than unimplemented functionality
- [ ]  "of <this/next/last> <year/month/week>"
- [ ]  "# days <before X/after X/ago>" currently parses to "# days after today"

**Comment by ChanceNCounter** [comment]
Thanks to [this](https://community.mycroft.ai/t/day-identification-is-not-working-correctly/7419/4) forum post by user tiredoftry for discovering this via asking about the weather

**Comment by JarbasAl** [comment]
i think the problem might be somewhere in the nice_date utils, doesn't sound like it is in extract_datetime because "yesterday" wasn't handled at all

fix for "yesterday" available in lingua_franca

**Comment by ChanceNCounter** [comment]
Suggest migrating this to Lingua Franca and editing to reflect the parts Jarbas has fixed, unless "yesterday" is regarded as critical enough to warrant porting that fix back here.

**Comment by forslund** [comment]
Good call. Yesterday we started pushing for getting lingua-franca used directly by core so it's definitely better to continue the work there.
